### PR TITLE
[Tests-Only] Add acceptance tests to verify mount configurations using the occ command

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance tests to verify mount  configurations using the occ command

## Related Issue

Related to #36704 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 1]
```diff
@@ -1251,6 +1251,58 @@ public function administratorDeletesFolder($folder) {
...
+		foreach ($commandOutput as $i) {
+			$createdLocalStorage[$i->mount_id] = \ltrim($i->mount_point, '/');
+		}
+		foreach ($createdLocalStorage as $key => $value) {
+			if ($value === $localStorage) {
+				$mount_id = $key;
+			}
+		}
+		if (!isset($mount_id)) {
+			throw new Exception("Id not found for folder to be verified");
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion
			throw new Exception("Id not found for local storage $localStorage to be verified");
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 2]
```diff
@@ -1251,6 +1251,58 @@ public function administratorDeletesFolder($folder) {
...
+		foreach ($filteredArray as $entry) {
+			$keyValue = \preg_split("/[:]/", $entry);
+			if (isset($keyValue[1])) {
+				$ResultArray[$keyValue[0]] = $keyValue[1];
+			} else {
+				$ResultArray[$keyValue[0]] = "";
+			}
+		}
+		foreach ($expectedInfo as $element) {
+			Assert::assertTrue($element == \array_map('trim', $ResultArray));
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion
			Assert::assertEquals($element, \array_map('trim', $ResultArray));
```

File: tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
+@cli @skipOnLDAP @local_storage
+Feature: verify mount configuration using occ command
+  As an admin
+  I want to verify mount configuration for created local storage
+  So that I can assure the status of created local storage
+
+  Scenario: delete local storage
```
Comment (phil-davis) [inline tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature]:
```suggestion
  Scenario: verify local storage
```

File: tests/acceptance/features/bootstrap/OccContext.php (hunk 2 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/OccContext.php]: @jasson99 can this change be made?

File: tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    And the administrator has created the local storage mount "new_local_storage"
+    And the administrator has uploaded file with content "file in local storage" to "/local_storage2/file-in-local-storage.txt"
+    And the administrator has uploaded file with content "new file" to "/new_local_storage/new-file"
+    When the administrator verifies the mount configuration for local storage "local_storage2" using the occ command
+    Then the following mount configuration information should be listed:
+      | status | code | message |
+      |   ok   | 0    |         |
+
+
+
```
Comment (dpakach) [inline tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature]: remove extra lines

File: tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
+@cli @skipOnLDAP @local_storage
+Feature: verify mount configuration using occ command
+  As an admin
+  I want to verify mount configuration for created local storage
+  So that I can assure the status of created local storage
+
+  Scenario: verify mount configuration for created local storage
```
Comment (dpakach) [inline tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature]: Also create a scenario where multiple local storages are created.

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 6]
```diff
@@ -1251,6 +1251,58 @@ public function administratorDeletesFolder($folder) {
...
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theAdministratorVerifiesTheMountConfigurationForLocalStorageUsingTheOccCommand($localStorage) {
+		$createdLocalStorage = [];
+		$this->listLocalStorageMount();
+		$commandOutput = \json_decode($this->featureContext->getStdOutOfOccCommand());
+		foreach ($commandOutput as $i) {
+			$createdLocalStorage[$i->mount_id] = \ltrim($i->mount_point, '/');
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/OccContext.php]: I dont think you need `$createdLocalStorage` variable here. You can just get mount_id from here directly

File: tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature]: fixed

File: tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/features/cliLocalStorage/verifyMountConfiguration.feature]: fixed

File: tests/acceptance/features/bootstrap/OccContext.php (hunk 6 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/OccContext.php]: fixed

File: tests/acceptance/features/bootstrap/OccContext.php (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/OccContext.php]: fixed

File: tests/acceptance/features/bootstrap/OccContext.php (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/OccContext.php]: fixed

**Review by phil-davis — APPROVED** [review]

**Pull request merged**